appscale/tools/agents/azure_agent.py

```diff
@@ -427,18 +434,82 @@ def create_linux_configuration(self, parameters):
     Returns:
         An instance of LinuxConfiguration
     """
+    is_autoscale = parameters['autoscale_agent']
     keyname = parameters[self.PARAM_KEYNAME]
     private_key_path = LocalState.LOCAL_APPSCALE_PATH + keyname
     public_key_path = private_key_path + ".pub"
+    auth_keys_path = "/home/{}/.ssh/authorized_keys".format(self.ADMIN_USERNAME)
+
+    if is_autoscale in ['True', True]:
+      public_key_path = auth_keys_path
+
     with open(public_key_path, 'r') as pub_ssh_key_fd:
       pub_ssh_key = pub_ssh_key_fd.read()
-    key_path = "/home/{}/.ssh/authorized_keys".format(self.ADMIN_USERNAME)
-    public_keys = [SshPublicKey(path=key_path, key_data=pub_ssh_key)]
+
+    public_keys = [SshPublicKey(path=auth_keys_path, key_data=pub_ssh_key)]
     ssh_config = SshConfiguration(public_keys=public_keys)
     linux_config = LinuxConfiguration(disable_password_authentication=True,
                                       ssh=ssh_config)
     return linux_config
 
+  def add_instances_to_existing_ss(self, count, parameters):
+    """ Looks through existing scale sets in a particular resource group and
+    adds instances (created as a part of autoscaling) to the ones which have
+    additional capacity.
+    Args:
+        count: The number of instances to be created for autoscaling.
+        parameters: A dict, containing all the parameters necessary to
+          authenticate this user with Azure.
+    Returns:
+        The number of instances created and added to the existing scale sets.
+    """
+    credentials = self.open_connection(parameters)
+    subscription_id = str(parameters[self.PARAM_SUBSCRIBER_ID])
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
+    compute_client = ComputeManagementClient(credentials, subscription_id)
+
+    num_instances_added = 0
+    vmss_list = compute_client.virtual_machine_scale_sets.list(resource_group)
+    for vmss in vmss_list:
+      vm_list = compute_client.virtual_machine_scale_set_vms.list(
+        resource_group, vmss.name)
+      ss_instances = []
```

Comment:
since we always use this for its length, would it be better to have a counter instead?